ports/stm32/eth.c

```diff
@@ -165,14 +167,14 @@ STATIC void eth_phy_write(uint32_t reg, uint32_t val) {
     }
     ETH->MACMIIDR = val;
     uint32_t ar = ETH->MACMIIAR;
-    ar = reg << ETH_MACMIIAR_MR_Pos | (ar & ETH_MACMIIAR_CR_Msk) | ETH_MACMIIAR_MW | ETH_MACMIIAR_MB;
+    ar = reg << ETH_MACMIIAR_MR_Pos | (ar & ETH_MACMIIAR_CR_Msk) | ETH_MACMIIAR_MW | ETH_MACMIIAR_MB | (phy_addr << ETH_MACMIIAR_PA_Pos);
```

Comment:
please put the pyh_addr bit at the start of this expression, to retain the ordering that most-significant-bits are on the left